src/react-components/profile-entry-panel.js

```diff
@@ -13,49 +13,76 @@ class ProfileEntryPanel extends Component {
   constructor(props) {
     super(props);
     window.store = this.props.store;
-    this.state = {name: this.props.store.state.profile.display_name};
+    this.state = {
+      display_name: this.props.store.state.profile.display_name,
+      avatar: this.props.store.state.profile.avatar,
+    };
     this.props.store.addEventListener("statechanged", this.storeUpdated);
   }
 
   storeUpdated = () => {
-    this.setState({name: this.props.store.state.profile.display_name});
+    this.setState({
+      display_name: this.props.store.state.profile.display_name,
+      avatar: this.props.store.state.profile.avatar,
+    });
   }
 
-  saveName = (e) => {
+  saveStateAndFinish = (e) => {
     e.preventDefault();
-    this.props.store.update({ profile: { display_name: this.nameInput.value } });
+    this.props.store.update({profile: {
+      display_name: this.state.display_name,
+      avatar: this.state.avatar
+    }});
     this.props.finished();
   }
 
+  stopPropagation = (e) => {
+    e.stopPropagation();
+  }
+
+  setAvatarStateFromIframeMessage = (e) => {
+    if (e.source !== this.avatarSelector.contentWindow) { return; }
+    this.setState({avatar: e.data.avatar});
+  }
+
   componentDidMount() {
     // stop propagation so that avatar doesn't move when wasd'ing during text input.
-    this.nameInput.addEventListener('keydown', e => e.stopPropagation());
-    this.nameInput.addEventListener('keypress', e => e.stopPropagation());
-    this.nameInput.addEventListener('keyup', e => e.stopPropagation());
+    this.nameInput.addEventListener('keydown', this.stopPropagation);
```

Comment:
On the fence if we need these. I believe since now a-frame has no input mappings until you join the room, it should work OK. however if we want to allow in-game identity changes then this should stay a-frame will be receiving input at that point.

Reply:
Yeah, I'll try removing these.

Reply:
I think it definitely makes sense to allow in game identity chagnes. That said we should probably switch to a different action set when we have the UI open instead of just blocking keyboard events.

Reply:
Ah, forgot about in-game identity changing. Er, I agree with switching action sets, but I'm going to leave that for later, this PR is big enough.